pandas/io/formats/csvs.py

```diff
@@ -128,19 +128,33 @@ def save(self):
         else:
             encoding = self.encoding
 
-        # PR 21300 uses string buffer to receive csv writing and dump into
-        # file-like output with compression as option. GH 21241, 21118
-        f = StringIO()
-        if not is_file_like(self.path_or_buf):
-            # path_or_buf is path
-            path_or_buf = self.path_or_buf
-        elif hasattr(self.path_or_buf, 'name'):
-            # path_or_buf is file handle
-            path_or_buf = self.path_or_buf.name
+        if self.compression and hasattr(self.path_or_buf, 'write'):
```

Comment:
can uou add a comment here (sure the warning says it all, but helpful nonetheless), also an issue reference